[WebRTC] Fix broken target for avif_decode_fuzzer

ddkilzer opened the pull request:
#### ac5eb110be5afb3b9ef5ca60842ca337b55b5a75
<pre>
[WebRTC] Fix broken target for avif_decode_fuzzer
<a href="https://bugs.webkit.org/show_bug.cgi?id=270936">https://bugs.webkit.org/show_bug.cgi?id=270936</a>
&lt;<a href="https://rdar.apple.com/124556987">rdar://124556987</a>&gt;

Reviewed by Said Abou-Hallawa.

* Source/WebCore/PAL/ThirdParty/libavif/libavif.xcodeproj/project.pbxproj:
- Clean up the project by removing many duplicate folders.
- Remove the android_jni folder since it was deleted in 275928@main.
(avif_decode_fuzzer target):
- Fix avif_decode_fuzzer by removing repro_fuzz.cc from the Sources
  list.  This fixes the issue described by the bug.
* Source/WebCore/PAL/ThirdParty/libavif/tests/oss-fuzz/avif_decode_fuzzer.cc:
(LLVMFuzzerTestOneInput):
- Add back declaration that was originally added in 272156@main, but
  removed in 275291@main.  This fixes a -Wmissing-prototypes compiler
  warning.

Canonical link: <a href="https://commits.webkit.org/276113@main">https://commits.webkit.org/276113@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da0eeaa7ba9cc9a4f2ecec556179bd94670eafee

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43666 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22714 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46104 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46309 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39789 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45970 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26539 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20122 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36073 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44240 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19729 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37599 "Passed tests") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17051 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; 1 api test failed or timed out; 1 api test failed or timed out; Running compile-webkit-without-change") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17277 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38661 "Found 1 new test failure: imported/w3c/web-platform-tests/xhr/setrequestheader-case-insensitive.htm (failure)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1723 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39823 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38968 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47860 "Built successfully") | 
| | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15294 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42868 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20122 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41544 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20303 "Built successfully") | | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5982 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19749 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/PAL/ThirdParty/libavif/tests/oss-fuzz/avif_decode_fuzzer.cc:
```diff
@@ -4,6 +4,8 @@
 #include "avif/avif.h"
 #include "avif/avif_cxx.h"
 
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t * Data, size_t Size);
```
Yes my submission [275291@main](https://commits.webkit.org/275291@main) removed this line. Sorry!

ddkilzer replied:
It's okay.  That introduced a warning, but for various reasons, `-Wno-error` is enabled on fuzzer builds.

The issue was adding `repro_fuzz.cc` to the compiled sources for the `avif_decoder_fuzzer` target, which didn't fail at build time, but failed at runtime.

shallawa reviewed (approved):
rs=me

Commit:
[WebRTC] Fix broken target for avif_decode_fuzzer
https://bugs.webkit.org/show_bug.cgi?id=270936
<rdar://124556987>

Reviewed by Said Abou-Hallawa.

* Source/WebCore/PAL/ThirdParty/libavif/libavif.xcodeproj/project.pbxproj:
- Clean up the project by removing many duplicate folders.
- Remove the android_jni folder since it was deleted in 275928@main.
(avif_decode_fuzzer target):
- Fix avif_decode_fuzzer by removing repro_fuzz.cc from the Sources
  list.  This fixes the issue described by the bug.
* Source/WebCore/PAL/ThirdParty/libavif/tests/oss-fuzz/avif_decode_fuzzer.cc:
(LLVMFuzzerTestOneInput):
- Add back declaration that was originally added in 272156@main, but
  removed in 275291@main.  This fixes a -Wmissing-prototypes compiler
  warning.

Canonical link: https://commits.webkit.org/276113@main
